rustls/src/client/hs.rs

```diff
@@ -683,71 +681,78 @@ impl State for ExpectServerHello {
             None
         };
 
-        // See if we're successfully resuming.
-        if let Some(ref resuming) = self.handshake.resuming_session {
-            if resuming.session_id == self.session_id {
-                debug!("Server agreed to resume");
-
-                // Is the server telling lies about the ciphersuite?
-                if resuming.supported_cipher_suite() != suite {
-                    let error_msg = "abbreviated handshake offered, but with varied cs".to_string();
-                    return Err(TlsError::PeerMisbehavedError(error_msg));
-                }
-
-                // And about EMS support?
-                if resuming.extended_ms != self.using_ems {
-                    let error_msg = "server varied ems support over resume".to_string();
-                    return Err(TlsError::PeerMisbehavedError(error_msg));
+        match (&self.handshake.resuming_session, self.session_id) {
```

Comment:
Above, let's also fix the nit Brian mentioned here: https://github.com/ctz/rustls/pull/584/files/9d4761acc86de30e02f8460364ca49413d937b82#r599254767. (Line 634 in the current state of your branch.)